core/src/main/java/org/testcontainers/utility/MountableFile.java

```diff
@@ -168,6 +170,14 @@ private String extractClassPathResourceToTempLocation(final String hostPath) {
         return tmpLocation.getAbsolutePath();
     }
 
+    private File createTempDirectory() {
+        try {
+            return Files.createTempDirectory(Paths.get(System.getProperty("java.io.tmpdir")), TESTCONTAINERS_TMP_DIR_PREFIX).toFile();
```

Comment:
Is there a reason you are not using `createTempDirectory(String prefix, FileAttribute<?>... attrs)`? I assume this should have the same effect.

Reply:
Yes, you are right, it is the same.

Reply:
Do you want to change the method call, or do we have to specifically cater for macOS after all?

Reply:
I would like to change the method call and change temporary dir for macOS.

What do you think about the following method?
```java
private File createTempDirectory() {
    try {
        if (SystemUtils.IS_OS_MAC) {
            return Files.createTempDirectory(Paths.get("/tmp"), TESTCONTAINERS_TMP_DIR_PREFIX).toFile();
        }
        return Files.createTempDirectory(TESTCONTAINERS_TMP_DIR_PREFIX).toFile();
    } catch  (IOException e) {
        return new File(TESTCONTAINERS_TMP_DIR_PREFIX + Base58.randomString(5));
    }
}
```

Reply:
In which case would the code inside the `catch` block be desired behavior I wonder?

Reply:
For example user runs java with params `-Djava.io.tmpdir=<folder>` on linux/windows.

In cases, the folder does not exist we will get IOException.

I think we could try to create a temp folder in the current folder as it was before